qdev_wrappers/T3/customised_instruments.py

```diff
@@ -227,78 +227,66 @@ def __init__(self, name, address, config, **kwargs):
-        self.dcbias.label = config.get('Decadac Channel Labels', dcbias_i)
-
-        # Assign custom variable names
-        lcut = config.get('Channel Parameters', 'left cutter')
-        self.lcut = self.channels[int(lcut)].volt
-
-        rcut = config.get('Channel Parameters', 'right cutter')
-        self.rcut = self.channels[int(rcut)].volt
-
-        jj = config.get('Channel Parameters', 'central cutter')
-        self.jj = self.channels[int(jj)].volt
-
-        rplg = config.get('Channel Parameters', 'right plunger')
-        self.rplg = self.channels[int(rplg)].volt
-
-        lplg = config.get('Channel Parameters', 'left plunger')
-        self.lplg = self.channels[int(lplg)].volt
-
-        self.add_parameter('cutters',
-                           label='{} cutters'.format(self.name),
-                           # use lambda for late binding
-                           get_cmd=self.get_cutters,
-                           set_cmd=self.set_cutters,
-                           unit='V',
-                           get_parser=float)
-
-        # Set up voltage and ramp safetly limits in software
-        ranges = config.get('Decadac Channel Limits')
-        ramp_settings = config.get('Decadac Channel Ramp Setttings')
 
+        '''
+        config file redesigned to have all channels for overview. Indices in config_settings[] for each channel are:
+        0: Channels name for deca.{}
+        1: Channel label
+        2: Channels unit (included as we are using decadac to control the magnet)
+        3: Voltage division factor
+        4: step size
+        5: delay
+        6: max value
+        7: min value
+        8: Fine or coarse mode channel
+        '''
+
+        # Couldnt get this to work with normal parameters so ended quite ugly using 'exec' to define input strings as methods
+        config_file = config.get('Decadac')
+        
         for chan in range(20):
-            try:
-                chan_range = ranges[str(chan)]
-            except KeyError:
-                continue
-            range_minmax = chan_range.split(" ")
-            if len(range_minmax) != 2:
-                raise ValueError(
-                    "Expected: min max. Got {}".format(chan_range))
-            else:
-                rangemin = float(range_minmax[0])
-                rangemax = float(range_minmax[1])
+            config_settings = config_file[str(chan)].split(",")
+            
+            step = float(config_settings[4])
+            delay = float(config_settings[5])
+            self.channels[chan].volt.set_step(step)
+            self.channels[chan].volt.set_delay(delay)
+            
+            
+            rangemin = float(config_settings[6])
+            rangemax = float(config_settings[7])
             vldtr = vals.Numbers(rangemin, rangemax)
             self.channels[chan].volt.set_validator(vldtr)
+            if config_settings[8] == 'fine':
+                if np.mod(chan,4)<3:
```

Comment:
Why do you exclude the last channel of every slot?

Reply:
that's a mistake. Should have been less than or equal to 3 (or <4). Haven't noticed as we have never used the last channel.

Reply:
But isn't mod(x,3)∈{0,1,2,3}? So that the if statement is always true?

Reply:
I take back what I said before. I think my idea was to not define the last two channels when in 'fine' mode but then made a wrong implementation. Also this isn't really needed as fine_volt already throws an error for the last two channels of each slot.
Maybe just remove the if statement for both 'coarse' and 'fine' mode? It definitely seems unnecessary.

Reply:
ok